firmware/library/L1_Drivers/ST7066U.hpp

```diff
@@ -0,0 +1,196 @@
+
+// The file contains the driver class for the ST7066U Parallel LCD Driver.
+//
+// Usage:
+//      Initialize the driver by calling the constructor:
+//       ST7066U(BusMode busMode, DisplayLines numLines,
+//               Font fontStyle, ControlPins_t *pins)
+//      If the desired busMode is 4-bits, then initialize D3, D2, D1,
+//      and D0 in ControlPins_t as NULL.
+//      Characters can be generated on the display using the function:
+//          void displayText(const char *text, CursorPosition_t pos)
+#pragma once
+
+#include "L0_LowLevel/LPC40xx.h"
+#include "L1_Drivers/gpio.hpp"
+
+class ST7066UInterface
+{
+ public:
+    struct CursorPosition_t;
+
+    virtual void init()                                               = 0;
+    virtual void clearDisplay()                                       = 0;
+    virtual void toggleDisplay(bool on)                               = 0;
+    virtual void toggleCursor(bool on)                                = 0;
+    virtual void setCursorDirection(bool increment)                   = 0;
+    virtual void displayText(const char * text, CursorPosition_t pos) = 0;
+
+ protected:
+    // Data transfer operation types.
+    enum class WriteOperationType : uint8_t
+    {
+        kCommandWrite = 0,
+        kDataWrite    = 1
```

Comment:
The word *Write* is a bit redundant here since the name of the class is WriteOperation. From ourside of the class one would write this: `WriteOperation::kData` or `WriteOperation::kCommand`. Since write is in the enum name, the "write" is understood.